code/game/turfs/turf.dm

```diff
@@ -579,9 +587,28 @@ GLOBAL_LIST_EMPTY(station_turfs)
 /turf/proc/acid_melt()
 	return
 
-/turf/rust_heretic_act()
+/// Check if the heretic is strong enough to rust this turf
+/turf/rust_heretic_act(mob/living/source)
+	var/heretic_rust_strength
+	if(IS_HERETIC(source))
+		var/datum/antagonist/heretic/heretic_data = source.mind.has_antag_datum(/datum/antagonist/heretic)
+		heretic_rust_strength = heretic_data.rust_strength
+	else if(IS_HERETIC_MONSTER(source))
+		var/datum/antagonist/heretic_monster/heretic_monster_data = source.mind.has_antag_datum(/datum/antagonist/heretic_monster)
+		var/datum/antagonist/heretic/master_heretic_data = heretic_monster_data.master.has_antag_datum(/datum/antagonist/heretic)
+		heretic_rust_strength = master_heretic_data.rust_strength
+	else if(source == null) // This should only apply to rust being spread by an ascended rust heretic's rune
+		heretic_rust_strength = 4
```

Comment:
Also, this if statement needs to be re-arranged a bit.

Check if source is null first (also, use the helper)
```dm
if(isnull(source))
   ...
```

`IS_HERETIC` actually returns the heretic antag datum if found - you can use this to save yourself some `has_antag_datum` calls.
```dm
var/datum/antagonist/heretic/heretic_data = IS_HERETIC(source)
if(heretic_data)
   ...
```